docs/ko/reference/promise/withTimeout.md

```diff
@@ -0,0 +1,48 @@
+# withTimeout
+
+지정한 시간보다 늦은 시간에 응답할 경우 `TimeoutError` 에러로 처리해요.
+
+이 함수는 특정한 시간 이후 보다 value의 응답이 늦게 올 경우 TimeoutError로 reject되는 Promise를 반환해요. async/await 함수를 사용하는 경우에 함수의 최대 실행 시간을 정할 수 있어요. 또한, 선택 옵션으로 지연을 취소할 수 있는 AbortSignal을 지원해요.
+
+## 인터페이스
+
+```typescript
+function withTimeout<T>(value: () => Promise<T>, ms: number, options?: WithTimeoutOptions): Promise<T>;
+```
+
+### 파라미터
+
+- `value` (`() => Promise<T>`): 실행할 Promise 값.
+- `ms` (`number`): Promise의 최대 실행 값을 지정할 밀리세컨드.
+- `options` (`WithTimeoutOptions`, optional): 옵션 객체.
+  - `signal` (`AbortSignal`, optional): timeout을 취소하기 위한 선택적 `AbortSignal`.
```

Comment:
I'm a bit confused about the `AbortSignal` in the `withTimeout` function. 

It's unclear whether the signal is for aborting the provided function or just the timeout itself, allowing the function to continue running. 

Is there a specific scenario where this would be useful?

Reply:
@raon0211 

I don't have any specific scenario in mind, but I added it because I thought it would be good to refer to `delay.ts`

Reply:
I guess this might make some kind of confusion, why not skip it this time and add it when it is needed?

Reply:
Our implementation would be much simpler.

```tsx
async function timeout(timeoutMs: number) {
  await delay(timeoutMs);
  throw new TimeoutError();
}

function withTimeout<T>(run: () => Promise<T>, timeoutMs: number) {
  return Promise.race([run(), timeout(timeoutMs)]);
}

Reply:
Good, remove `AbortSignal`, and we're going to prepare `timeout` and `withTimeout`, and docs.